schemas/config/3.0.schema.json

```diff
@@ -1,2600 +1,2869 @@
+									"type": "string",
+									"description": "Metrics type",
+									"anyOf": [
+										{
+											"type": "string",
+											"examples": [
+												"summary",
+												"histogram"
+											]
+										}
+									]
+								},
+								"prefix": {
+									"description": "Temporal metrics prefix",
+									"type": "string",
+									"default": null
+								}
+							}
+						},
+						{
+							"properties": {
+								"host_port": {
+									"description": "The host and port of the statsd server",
+									"type": "string",
+									"default": "127.0.0.1:8125"
+								},
+								"prefix": {
+									"description": "The prefix to use in reporting to statsd",
+									"type": "string",
+									"default": null
+								},
+								"flush_interval": {
+									"description": "FlushInterval is the maximum interval for sending packets",
+									"type": "string",
+									"default": "1s"
+								},
+								"flush_bytes": {
+									"description": "FlushBytes specifies the maximum udp packet size you wish to send. If FlushBytes is unspecified, it defaults  to 1432 bytes, which is considered safe for local traffic.",
+									"type": "integer",
+									"default": 1432
+								},
+								"tags": {
+									"description": "Hashmap with tag:value values",
+									"$ref": "#/definitions/HashmapString"
+								},
+								"tag_prefix": {
+									"description": "Prefix for the tag",
+									"type": "string",
+									"default": null
+								},
+								"tag_separator": {
+									"description": "TagSeparator allows tags to be appended with a separator. If not specified tag keys and values are embedded to the stat name directly.",
+									"type": "string",
+									"default": null
+								}
+							}
+						}
+					]
+				},
+				"activities": {
+					"description": "Activities pool settings",
+					"type": "object",
+					"$ref": "#/definitions/WorkersPool"
+				},
+				"tls": {
+					"description": "Temporal TLS configuration",
+					"type": "object",
+					"required": [
+						"key",
+						"cert"
+					],
+					"properties": {
+						"key": {
+							"$ref": "#/definitions/TLSKeyFile"
+						},
+						"cert": {
+							"$ref": "#/definitions/TLSCertFile"
+						},
+						"root_ca": {
+							"$ref": "#/definitions/TLSCAFile"
+						},
+						"client_auth_type": {
+							"$ref": "#/definitions/TLSClientAuthType"
+						},
+						"server_name": {
+							"description": "ServerName is used to verify the hostname on the returned certificates unless InsecureSkipVerify is given. It is also included in the client's handshake to support virtual hosting unless it is an IP address.",
+							"type": "string"
+						}
+					}
+				}
+			}
+		},
+		"kv": {
+			"description": "Key value storage plugin.",
+			"type": "object",
+			"minProperties": 1,
+			"patternProperties": {
+				"[a-zA-Z0-9_-]*": {
+					"description": "The name of the key-value storage, as used in your application.",
```

Comment:
Created if-else here so options validate correctly depending on the selected `driver`.